packages/core/src/State.ts

```diff
@@ -335,21 +335,32 @@ export class State<
 
     const transitionData = this.machine?.getTransitionData(this, event);
 
+    if (!transitionData) {
+      return false;
+    }
+
     /**
      * The state will change from this event if:
      * - There are entry actions
      * - There are exit actions
-     * - There are transitions in which either the target has changed or there are transition actions
+     * - There are transitions in which either the targets have changed or there are transition actions
      */
-    return !transitionData
-      ? false
-      : transitionData.entrySet.some((sn) => sn.onEntry.length > 0) ||
-          transitionData.exitSet.some((sn) => sn.onExit.length > 0) ||
-          !!transitionData?.transitions.some((t) => {
-            return (
-              (t.target && t.target.length > 0 && t.target[0] !== t.source) ||
-              t.actions.length > 0
-            );
-          });
+    if (
```

Comment:
resolving this correctly is quite complex, just added yet another failing test with this logic: https://github.com/statelyai/xstate/pull/2850

I wonder if it wouldn't be just easier to redefine `state.can` as "will transition be selected based on the given event". This way we'd only have to check something like:
```js
return !!transitionData?.getTransitionData(this, event).transitions.length
```

Reply:
@Andarist Changed the behavior here: 9c1a28e